File: Java/RecursionPractice/src/calculations/FibonacciTest.java
package calculations;

import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.*;

class FibonacciTest {

    Fibonacci test1;

    Fibonacci test2;

    Fibonacci test3;

    @BeforeEach
    void setUp() {
        this.test1 = new Fibonacci(3);
        this.test2 = new Fibonacci(8);
        this.test3 = new Fibonacci(0);
    }

    @Test
    void getValue() {
        assertEquals(this.test1.getValue(), 2);
        assertEquals(this.test2.getValue(), 21);
        assertEquals(this.test3.getValue(), 0);
    }
}

File: Kotlin/RecursionPractice/test/calculations/FibonacciTest.kt
package calculations

import org.junit.jupiter.api.Test

import org.junit.jupiter.api.Assertions.*

class FibonacciTest {

    private val test1 = Fibonacci(3)

    private val test2 = Fibonacci(8)

    private val test3 = Fibonacci(0)

    @Test
    fun getValue() {
        assertEquals(test1.getValue(), 2)
        assertEquals(test2.getValue(), 21)
        assertEquals(test3.getValue(), 0)
    }
}